erc20/src/lib.rs

```diff
@@ -81,25 +98,17 @@ impl ERC20 {
         true
     }
 
+    // -- GETTER FUNCTIONS ----------------------------------------------------
+
     pub fn total_supply(&self) -> U256 {
-        self.total_supply
+        self.total_supply.read()
```

Comment:
I think you wrote this somewhere, but `pub` fields could also get automatic getters, can be done in the future. (let's open an issue)